doc/design/audit-trail-mvp.md

```diff
@@ -0,0 +1,152 @@
+# Audit Trail MVP
+
+NIH has required that DANDI Archive generate audit logs while leaving undefined
+what that means. We are choosing to fulfill the requirement with a feature that
+indicates to Dandiset owners and admins the sequence of significant actions
+taken to modify Dandisets, along with identification of who performed each
+action.
+
+The major motivation is to inform owners about how their Dandisets have changed
+when a given owner did not make the change and wishes to know who to speak to
+about those changes. In essence, it is a tool to support effective
+collaboration.
+
+## Requirements
+
+1. **Persistent tracking of significant actions.** Whenever a *significant
+   action* (see list [below](#significant-actions)) occurs successfully, the archive will generate a
+   persistent record of that action, including the following data:
+    - The date and time
+    - The authorized user who triggered the action
+    - The ID of the Dandiset on which the action occurs
+    - Event-specific data about the action itself
+
+2. **Display of audit log (to owners and admins only).** The Dandiset landing
+   page will display a rendered (i.e. human-readable) version of the log event
+   data (as long as the viewing user is an owner of the Dandiset, or an admin).
```

Comment:
Thanks @waxlamp.  It might be useful for the audit logs to be public to all users.  For users performing downstream analysis, they might be interested in understanding the changes between versions.  Similar to being able to traverse a git commit history. 
 Is there a privacy reason that we wouldn't want to make the logs public?  Perhaps this suggestion would be beyond the MVP?

Reply:
We have a feature in the design phase (well, pre-design phase, meaning it's on our minds) to enable owners to bundle release notes with a published Dandiset version.

I would argue that we need to keep this audit log private for now, until we have a chance to analyze the privacy considerations of making it public, as well as determining exactly why we would want to make this public. As conceived currently, this is a mechanism for enabling communication and collaboration between Dandiset owners, so conservatively we're keeping this private. Put another way, "might be useful" doesn't feel a strong enough reason to make this public at the moment.

Reply:
Thanks @waxlamp.  I agree.